#! /bin/sh /usr/share/dpatch/dpatch-run
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: 30_scripts__mysqld_safe.sh__signals.patch by  <ch@debian.org>

@DPATCH@
diff -urNad '--exclude=CVS' '--exclude=.svn' '--exclude=.git' '--exclude=.arch' '--exclude=.hg' '--exclude=_darcs' '--exclude=.bzr' Percona-Server~/scripts/mysqld_safe.sh Percona-Server/scripts/mysqld_safe.sh
--- Percona-Server~/scripts/mysqld_safe.sh	2013-03-04 08:26:59.000000000 -0500
+++ Percona-Server/scripts/mysqld_safe.sh	2013-03-04 08:27:48.000000000 -0500
@@ -29,9 +29,6 @@
 syslog_tag_mysqld=mysqld
 syslog_tag_mysqld_safe=mysqld_safe
 
-trap '' 1 2 3 15			# we shouldn't let anyone kill us
-trap '' 13                              # not even SIGPIPE
-
 # MySQL-specific environment variable. First off, it's not really a umask,
 # it's the desired mode. Second, it follows umask(2), not umask(3) in that
 # octal needs to be explicit. Our shell might be a proper sh without printf,
@@ -739,6 +736,12 @@
   fi
 fi
 
+# From now on, we catch signals to do a proper shutdown of mysqld
+# when signalled to do so.
+#
+trap '@bindir@/mysqladmin --defaults-extra-file=/etc/mysql/debian.cnf refresh' 1 # HUP
+trap '@bindir@/mysqladmin --defaults-extra-file=/etc/mysql/debian.cnf shutdown' 2 3 13 15 # INT QUIT PIPE and TERM
+
 #
 # Uncomment the following lines if you want all tables to be automatically
 # checked and repaired during startup. You should add sensible key_buffer
